### ISSUE TITLE
list_all output has changed

### ISSUE THREAD

**Issue by klp2** [glasswalk3r/archive-tar-wrapper-perl#4]
Hi there

It looks like `tar_path` that is returned for each item in `list_all` is now prefixed by `tar/`. Is that expected to stay going forward?

**Comment by oschwald** [comment]
This issue is causing the test suite for `Archive::Tar::Wrapper` to fail on installation:

```
#   Failed test 'Check list'
#   at t/001Basic.t line 42.
#          got: '001Basic.t foo/bar/baz foo/bar/permtest foo/bar/string tar/001Basic.t tar/foo/bar/baz tar/foo/bar/permtest tar/foo/bar/string'
#     expected: '001Basic.t foo/bar/baz foo/bar/permtest foo/bar/string'

#   Failed test 'foo excluded'
#   at t/001Basic.t line 159.
#          got: 'tar/boo/bar/baz'
#     expected: 'boo/bar/baz'
# Looks like you planned 24 tests but ran 25.
# Looks like you failed 2 tests of 25 run.
t/001Basic.t ..... 
Dubious, test returned 2 (wstat 512, 0x200)

```

**Comment by glasswalk3r** [comment]
No @klp2 , it's a bug, not a feature. It was introduced in release 0.26, due refactorings made.
Unfortunately, I wasn't able so far to reproduce the issue on my boxes.

 @oschwald , the tests aren't failing everywhere.

The current tests available are unable to detect such issue too, so besides the bugs, it is necessary to review why the tests didn't caught such thing. The bug is around `list_reset()` and `list_next()` functions, but I don't know what triggers it.

I'm using Ubuntu 16.04 and wasn't able to reproduce the issue on it too. I'm trying right now to check on Debian Jenny to see if I can reproduce the failures.

Based on [CPAN Testers Matrix](http://matrix.cpantesters.org/?dist=Archive-Tar-Wrapper+0.26) most of the errors are coming from Linux, but less than half are failing. I was able to identify two tests coming from the same box, one that passed and the other that failed.

So yes, I'm in need to receive more data about that. I did some changes in the code right now to be able to fetch more information, so if any one you could `git clone` the repository and give it a try with:

`$ perl -Ilib t/001Basic.t`

Inside the repository directory, it would fetch more information that could help me understand better the issue. Don't run the tests with `prove`, because I'm using `Log::Log4perl` to generate this additional information and that will mess with the TAP output.

**Comment by glasswalk3r** [comment]
I believe I fixed the issue with commit 5b12efd0be640377cb97807a381da438c2875689. Please check and let me know.

**Comment by glasswalk3r** [comment]
Given the [results](http://matrix.cpantesters.org/?dist=Archive-Tar-Wrapper+0.27) from the release 0.27, I'm closing this one.
